### ISSUE TITLE
When the file casing differs between SARIF file/log and the filesystem, issues do not populate the Problems Pane

### ISSUE THREAD

**Issue by HenokAddis** [microsoft/sarif-vscode-extension#323]
When launching the SARIF Viewer automatically after a Code Analysis build, you can see that multiple SARIF files have been generated with issues.

![image](https://user-images.githubusercontent.com/46505299/89837427-cc84a100-db1d-11ea-85d4-e8097264879f.png)

However after closer observation, the SARIF files which are not identical to their file-system file casing do not populate in the problems pane.
![image](https://user-images.githubusercontent.com/46505299/89837501-f0e07d80-db1d-11ea-9048-74cc828e671f.png)

**Comment by GabeDeBacker** [comment]
This was originally addressed with this pull-request #208. 

URI's are case sensitive and VSCode treats them this way regardless of the scheme.

Now, the analyzers running against the source for Windows lower-cases all URIs written into the SARIF file, and the viewer does not modify the casing of any URIs it sees and also treats them as URIs (i.e. case-sensitive).

As a result, the URIs do not compare against what VSCode has and this problem results.

Arguably what should be happening is that the URIs in the SARIF logs should be case-correct. This would require changes to the Windows analyzers. In the meantime, the SARIF viewer could perform a check to see if the OS platform is Windows and fix the URI file scheme paths (if possible) when the log file is opened.

This will cause a performance penalty on the Windows platform as it will cause a file-system enumeration for every URI that is encountered.

**Comment by HenokAddis** [comment]
`This will cause a performance penalty on the Windows platform as it will cause a file-system enumeration for every URI that is encountered.`

@jeffersonking - Could we measure what this performance hit would look like? This seems like a ship blocker to broader audiences

**Comment by jeffersonking** [comment]
I have a tactical fix I'm testing. It should unblock the majority of your cases while giving us some breathing room to think about the longer-term strategy.

**Comment by HenokAddis** [comment]
TLDR - 

1. VSCode does not accept only lowercased URIs so simply lowercasing everything will not work
so file:///c:/myFile.cpp is different than file:///c:/myfile.cpp
2. We make subsequent calls to Build.exe and NMake and we need to investigate both where this fix needs to go, and who needs to own it.
3. Here is proof that the build system does this lowercasing
3001>'s:\repos\os\src\tools\amd64\nmake.exe /nologo BUILDMSG=Stop. analyze /nologo /f s:\repos\os\src\tools\makefile.def BUILD_PASS=PASS1 NOLINK=1 MAKEDIR_RELATIVE_TO_BASEDIR=sdktools\debuggers\imgutil\binplace TARGET_PLATFORM=windows
3001>filemappingfuncs.c

Which is not the correct casing for:

os\src\sdktools\debuggers\imgutil\binplace\FileMappingFuncs.c.

**Comment by HenokAddis** [comment]
I believe Jeff's solution will not be pursued either

**Comment by Miaku** [comment]
@jeffersonking has a temporary fix up for this issue. @HenokAddis and @Miaku to test this temporary fix.

**Comment by jeffersonking** [comment]
Verified the common case with @Miaku and @HenokAddis.
